database/schema-user-preferences.sql

```diff
@@ -0,0 +1,119 @@
+-- NOTE: This schema depends on schema-friends
+
+-- psql -d pacific -f database/schema-user-preferences.sql
+
+-- Connect to database
+\c pacific
+
+-- Drop tables
+drop table if exists preferences_steps cascade;
+drop table if exists preferences cascade;
+drop table if exists users_preferences cascade;
+drop table if exists locations_types cascade;
+drop table if exists locations cascade;
+
+-- Create tables
+create table preferences_steps (
+  id          bigserial not null primary key,
+  label       varchar(50) not null,
+  question    varchar(255) not null,
+  order       integer not null
+);
+
+create table preferences (
+  id           bigserial not null primary key,
+  preference_step_id  bigserial not null,
+  parent_preference_id bigserial null,
+  label        varchar(50) not null,
+  description  varchar(255) null,
+  foreign key (preference_step_id) references preferences_steps(id),
+  foreign key (parent_preference_id) references preferences(id),
+);
+
+create table users_preferences (
+  user_id  bigserial not null,
+  preferences_id  bigserial not null,
+  foreign key (user_id) references users(id),
+  foreign key (preferences_id) references preferences(id)
+);
+
+create table locations_types (
+  id  bigserial not null primary key,
+  label varchar(255) null,
+);
+
+create table locations (
+  id        bigserial not null primary key,
+  location_type_id  bigserial not null,
+  label     varchar(255) null,
+  foreign key (location_type_id) references locations_types(id),
+);
+
+
+alter table users add location_id bigserial not null;
+alter table users add constraint  foreign key (location_id) references locations(id);
```

Comment:
@fernanda1411 - Since this `users` table is created in my `schema-friends.sql`, I can modify the table creation query in my file to include this field/foreign key. That way everything is in one place. I can submit a PR and tag you if that sounds good to you.

Reply:
I created PR #8 if you want to remove line 53 in your commit. You would still need to keep the foreign key creation in your file (line 54).

Later on we can probably combine the schema files into one.

Reply:
Sure. We can do it that way. There is not much difference on the DB creating it all at once versus modifying the table. 

I think it would be even more organized if we had the user schema in a separate file as it's not intuitive for look for it in `schema-friends.sql`. I'll add that as comment on your new PR.

Also, I followed your style on the backend, but I think we should have a `controllers` folder, a `models` folder and a `routes` folder but considering the time that we have for this project I've choose to just follow what you did.

Reply:
@awhlam  I've comment the changes related to users on my schema. 

```
-- Update users rows and update user 4 with a location preference.
-- As per PR feedback this will be implemented on a different schema.
-- alter table users add location_id bigint not null;
-- alter table users add constraint user_location_fk foreign key (location_id) references locations(id);
-- UPDATE users SET location_id = 1 WHERE id = 4; -- Update user Feranda with the location San Francisco
```

Reply:
Thanks @fernanda1411, I updated my PR #8